src/cargo/util/config.rs

```diff
@@ -154,6 +154,21 @@ impl Config {
         }
     }
 
+    pub fn get_path(&self, key: &str) -> CargoResult<Option<PathBuf>> {
+        if let Some((string, path)) = try!(self.get_string(&key)) {
+            let string_as_path = Path::new(&string);
+            if string_as_path.is_absolute() || string_as_path.components().count() == 1 {
+                Ok(Some(PathBuf::from(string_as_path)))
+            } else {
+                let prefix_path = try!(path.parent().and_then(|p| p.parent())
+                                       .ok_or(internal("unexpected config file path")));
+                Ok(Some(prefix_path.join(string_as_path)))
+            }
```

Comment:
Was the prior logic not sufficient? I don't think this needs to test for `is_absolute` at least because `join` already handles that and the component count often takes many other factors into account, so checking for slashes in theory should be enough.

Also, it's fine to `unwrap()` here as the name of all current config files is `.cargo/config` so they're guaranteed to have at least two parents.